smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/integration/DefaultReadmeGenerator.java

```diff
@@ -0,0 +1,102 @@
+import software.amazon.smithy.typescript.codegen.TypeScriptCodegenContext;
+import software.amazon.smithy.typescript.codegen.TypeScriptSettings;
+import software.amazon.smithy.typescript.codegen.TypeScriptWriter;
+import software.amazon.smithy.utils.IoUtils;
+import software.amazon.smithy.utils.SmithyInternalApi;
+import software.amazon.smithy.utils.StringUtils;
+
+@SmithyInternalApi
+public final class DefaultReadmeGenerator implements TypeScriptIntegration {
+
+    public static final String README_FILENAME = "README.md";
+    public static final String DEFAULT_CLIENT_README_TEMPLATE = "default_readme_client.md.template";
+    public static final String DEFAULT_SERVER_README_TEMPLATE = "default_readme_server.md.template";
+
+    @Override
+    public void customize(TypeScriptCodegenContext codegenContext) {
+        TypeScriptSettings settings = codegenContext.settings();
+
+        if (!settings.createDefaultReadme()) {
+            return;
+        }
+
+        String file = settings.generateClient() ? DEFAULT_CLIENT_README_TEMPLATE : DEFAULT_SERVER_README_TEMPLATE;
+
+        Model model = codegenContext.model();
+
+        codegenContext.writerDelegator().useFileWriter(README_FILENAME, "", writer -> {
+            ServiceShape service = settings.getService(model);
+            String resource =  IoUtils.readUtf8Resource(getClass(), file);
+            resource = resource.replaceAll(Pattern.quote("${packageName}"), settings.getPackageName());
+
+            String clientName = StringUtils.capitalize(service.getId().getName(service));
+
+            resource = resource.replaceAll(Pattern.quote("${serviceId}"), clientName);
+
+            String rawDocumentation = service.getTrait(DocumentationTrait.class)
+                    .map(DocumentationTrait::getValue)
+                    .orElse("");
+            String documentation = Arrays.asList(rawDocumentation.split("\n")).stream()
+                    .map(StringUtils::trim)
+                    .collect(Collectors.joining("\n"));
+            resource = resource.replaceAll(Pattern.quote("${documentation}"), Matcher.quoteReplacement(documentation));
+
+            TopDownIndex topDownIndex = TopDownIndex.of(model);
+            OperationShape firstOperation = topDownIndex.getContainedOperations(service).iterator().next();
+            String operationName = firstOperation.getId().getName(service);
+            resource = resource.replaceAll(Pattern.quote("${commandName}"), operationName);
+            resource = resource.replaceAll(Pattern.quote("${operationName}"),
+                    operationName.substring(0, 1).toLowerCase() + operationName.substring(1));
+
+            writer.write(resource.replaceAll(Pattern.quote("$"), Matcher.quoteReplacement("$$")));
+            writeOperationList(writer, model, settings);
+        });
+    }
+
+    private void writeOperationList(TypeScriptWriter writer, Model model, TypeScriptSettings settings) {
```

Comment:
What is this? Is this just for client or server or both? It is adding something to the end of the file, after the License?

Reply:
It was supposed to be a section which list out all the operations for the SDK. I have removed it for now.